### PR TITLE
Admin orders list

### PR DESCRIPTION
**Pull request by lcko1012**
## Related Tickets
- [#46928](https://edu-redmine.sun-asterisk.vn/issues/46928)

## WHAT (optional)
- Create orders list page for admin
- On orders admin page, can search by status

## HOW
- Use namespace for admin routes, check user is admin in `admin_controller.rb` file
- On orders admin page, use `scope` for status searching feature
- Besides that, `I18n` is used for the `status` attribute

## Evidence (Screenshot or Video)
image
image

### CODE REVIEW

File: app/controllers/admin/orders_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Admin::OrdersController < Admin::AdminController
+  def index
+    @q = Order.ransack(params[:q])
```
Comment (trinhdvt) [inline app/controllers/admin/orders_controller.rb]:
```suggestion
    @q = Order.ransack params[:q]
```

File: app/models/order.rb [hunk 2]
```diff
@@ -9,4 +9,12 @@ class Order < ApplicationRecord
...
     rejected: 4
   }, _prefix: true
+
+  scope :newest, ->{order created_at: :desc}
+
+  def self.status_attributes_for_select
+    statuses.map do |status, id|
+      [I18n.t("statuses.#{status}"), id]
+    end
+  end
```
Comment (trinhdvt) [inline app/models/order.rb]:
```suggestion
  class << self
    def status_attributes_for_select
      statuses.map do |status, id|
      [I18n.t("statuses.#{status}"), id]
    end
  end
```

File: app/views/admin/orders/_order.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+ <tr>
+    <td><%= order.receiver_name %></td>
+    <td><%= order.phone %></td>
+    <td>
+        <span class="order-status bg-<%= status_color order.status %>">
+            <%= t "statuses.#{order.status}" %>
+        </span>
+    </td>
+    <td><%= number_to_currency(order.total) %></td>
```
Comment (trinhdvt) [inline app/views/admin/orders/_order.html.erb]:
```suggestion
    <td><%= number_to_currency order.total %></td>
```

File: config/routes.rb [hunk 4]
```diff
@@ -10,5 +10,10 @@
     resources :account_activations, only: :edit
     resources :products, only: :show
     resources :carts, only: :create
+
+    namespace :admin do
+      root "orders#index"
+      resources :orders, only: [:index]
```
Comment (trinhdvt) [inline config/routes.rb]:
```suggestion
      resources :orders, only: :index
```

**Review by trinhdvt — CHANGES_REQUESTED** [review]
cmt

File: app/controllers/admin/orders_controller.rb (hunk 1 above)
Comment (lcko1012) [reply app/controllers/admin/orders_controller.rb]: fix

**Review by trinhdvt — APPROVED** [review]
ok

**Comment by lcko1012** [comment]
ready

**Review by trinhdvt — APPROVED** [review]
ok

File: app/views/admin/orders/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<%= provide :title, t(".title") %>
+
+<div class="admin-orders">
+  <h1><%= t ".title" %></h1>
+
+  <div class="search-wrapper">
+    <%= form_tag admin_orders_path, class: "search-form", :method => 'get' do %>
```
Comment (ghost) [inline app/views/admin/orders/index.html.erb]:
```suggestion
    <%= form_tag admin_orders_path, class: "search-form", method: :get do %>
```

File: app/views/admin/orders/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      <%= label_tag :status_id, t(".status") %>
+
+      <%= select_tag :status_id,
+          options_for_select(Order.status_attributes_for_select,
+          :selected =>params[:option]),
```
Comment (ghost) [inline app/views/admin/orders/index.html.erb]:
```suggestion
          selected: params[:option]),
```

File: app/models/order.rb [hunk 7]
```diff
@@ -9,4 +9,15 @@ class Order < ApplicationRecord
     received: 3,
     rejected: 4
   }, _prefix: true
+
+  scope :newest, ->{order created_at: :desc}
+  scope :status_search, ->(status_id){where("status = ?", status_id)}
```
Comment (ghost) [inline app/models/order.rb]:
```suggestion
  scope :status_search, ->(status_id){where("status = ?", status_id) if status_id.present?}
```

File: app/controllers/admin/orders_controller.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+class Admin::OrdersController < Admin::AdminController
+  def index
+    status_id = params[:status_id]
+    @orders = if status_id.present?
+                Order.status_search(status_id).newest
+              else
+                Order.newest
+              end
```
Comment (ghost) [inline app/controllers/admin/orders_controller.rb]:
```suggestion
    @orders = Order.status_search(status_id).newest
```

**Review by ghost — COMMENTED** [review]
cmt

File: app/controllers/admin/admin_controller.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+class Admin::AdminController < ApplicationController
+  before_action :require_admin
+  layout "admin"
```
Comment (ghost) [inline app/controllers/admin/admin_controller.rb:3]:
```suggestion
  layout :admin
```
Để như này dc ko e

**Review by ghost — COMMENTED** [review]
cmt

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**